QtCollider/widgets/QcWebView.cpp

```diff
@@ -20,126 +20,226 @@
+    for (int val : data) {
+      byteData.push_back((char)val);
+}
+    page()->setContent(byteData, mimeType, baseUrl);
+  }
 }
 
-int WebView::linkDelegationPolicy () const
+void WebView::toHtml(QcCallback* cb) const
 {
-  return (int)page()->linkDelegationPolicy();
+  if (page()) {
+    if (cb) {
+      page()->toHtml(cb->asFunctor());
+    } else {
+      page()->toHtml([](const QString&){});
+}
+  } else {
+    cb->asFunctor()(QString());
+  }
 }
 
-void WebView::setLinkDelegationPolicy ( int p )
+void WebView::toPlainText(QcCallback* cb) const
 {
-  page()->setLinkDelegationPolicy( (QWebPage::LinkDelegationPolicy)p );
+  if (page()) {
+    if (cb) {
+      page()->toPlainText(cb->asFunctor());
+    } else {
+      page()->toPlainText([](const QString&){});
+}
+  } else {
+    cb->asFunctor()(QString());
+  }
 }
 
-bool WebView::delegateReload() const
+void WebView::runJavaScript(const QString& script, QcCallback* cb)
 {
-  WebPage *p = qobject_cast<QtCollider::WebPage*>(page());
-  Q_ASSERT(p);
-  return p->delegateReload();
+  if (page()) {
+    if (cb) {
+      page()->runJavaScript(script, cb->asFunctor());
+    } else {
+      page()->runJavaScript(script, [](const QVariant&){});
+}
+  } else {
+    cb->asFunctor()(QString());
+  }
 }
 
-void WebView::setDelegateReload( bool flag )
+void WebView::setWebAttribute(int attr, bool on)
 {
-  WebPage *p = qobject_cast<QtCollider::WebPage*>(page());
-  Q_ASSERT(p);
-  p->setDelegateReload( flag );
+  if (page()) {
+    page()->settings()->setAttribute((QWebEngineSettings::WebAttribute)attr, on);
+}
 }
 
-void WebView::setFontFamily( int generic, const QString & specific )
+bool WebView::testWebAttribute(int attr)
 {
-  settings()->setFontFamily( (QWebSettings::FontFamily) generic, specific );
+  return page() ?
+    page()->settings()->testAttribute((QWebEngineSettings::WebAttribute)attr)
+    : false;
 }
 
-void WebView::evaluateJavaScript ( const QString &script )
+void WebView::resetWebAttribute(int attr)
 {
-  if( script.isEmpty() ) return;
-  QWebFrame *frame = page()->currentFrame();
-  if( frame ) frame->evaluateJavaScript( script );
+  if (page()) {
+    page()->settings()->resetAttribute((QWebEngineSettings::WebAttribute)attr);
+}
 }
 
-void WebView::findText( const QString &searchText, bool reversed )
+void WebView::navigate(const QString& urlString)
 {
-  QWebPage::FindFlags flags( QWebPage::FindWrapsAroundDocument );
-  if( reversed ) flags |= QWebPage::FindBackward;
-  QWebView::findText( searchText, flags );
+  QUrl url(urlString);
+  this->load(url);
 }
 
-void WebView::onLinkClicked( const QUrl &url )
+void WebView::findText( const QString &searchText, bool reversed, QcCallback* cb )
 {
-  Q_EMIT( linkActivated( url.toString() ) );
+  QWebEnginePage::FindFlags flags;
+  if( reversed ) flags |= QWebEnginePage::FindBackward;
+
+  if (!cb) {
+    QWebEngineView::findText(searchText, flags);
+  } else {
+    QWebEngineView::findText(searchText, flags, cb->asFunctor());
+}
```

Comment:
misaligned brace, also consider switching the condition so that the positive branch is first